core/vdbe/mod.rs

```diff
@@ -1011,6 +1028,196 @@ impl Program {
                     }
                     state.pc += 1;
                 }
+                Insn::BitAnd { lhs, rhs, dest } => {
+                    let lhs = *lhs;
+                    let rhs = *rhs;
+                    let dest = *dest;
+                    match (&state.registers[lhs], &state.registers[rhs]) {
+                        // handle 0 and null cases
+                        (OwnedValue::Null, _) | (_, OwnedValue::Null) => {
+                            state.registers[dest] = OwnedValue::Null;
+                        }
+                        (_, OwnedValue::Integer(0))
+                        | (OwnedValue::Integer(0), _)
+                        | (_, OwnedValue::Float(0.0))
+                        | (OwnedValue::Float(0.0), _) => {
+                            state.registers[dest] = OwnedValue::Integer(0);
+                        }
+                        (OwnedValue::Integer(lh), OwnedValue::Integer(rh)) => {
+                            state.registers[dest] = OwnedValue::Integer(lh & rh);
+                        }
+                        (OwnedValue::Float(lh), OwnedValue::Float(rh)) => {
+                            state.registers[dest] = OwnedValue::Integer(*lh as i64 & *rh as i64);
+                        }
+                        (OwnedValue::Float(lh), OwnedValue::Integer(rh)) => {
+                            state.registers[dest] = OwnedValue::Integer(*lh as i64 & rh);
+                        }
+                        (OwnedValue::Integer(lh), OwnedValue::Float(rh)) => {
+                            state.registers[dest] = OwnedValue::Integer(lh & *rh as i64);
+                        }
+                        (OwnedValue::Agg(aggctx), other) | (other, OwnedValue::Agg(aggctx)) => {
```

Comment:
Not a comment on your PR, but on the code overall in this module: we really need to abstract this aggctx handling somehow, there's so much repetition... anyway, carry on.